### PR TITLE
port to 19w07a

### PR DESCRIPTION
**Pull request by asiekierka**

### CODE REVIEW

**Comment by asiekierka** [comment]
I'm a bit torn about the Identifiable/Simple prefixes. Maybe we can use something else?

File: src/main/java/net/fabricmc/fabric/api/resource/SimpleResourceReloadListener.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * the apply stage is guaranteed to run on the game thread.
+ *
+ * For a fully synchronous alternative, consider using
+ * {@link SynchronousResourceReloadListener} in conjunction with
+ * {@link IdentifiableResourceReloadListener}.
+ *
+ * @param <T> The data object.
+ */
+public interface SimpleResourceReloadListener<T> extends IdentifiableResourceReloadListener {
+	default CompletableFuture<Void> apply(ResourceReloadListener.Helper var1, ResourceManager var2, Profiler var3, Profiler var4, Executor var5, Executor var6) {
```
Comment (SquidDev) [inline src/main/java/net/fabricmc/fabric/api/resource/SimpleResourceReloadListener.java]: Can we pick better variable names here, or is it not worth it seeing as it'll (probably) change next snapshot?

Comment (asiekierka) [reply src/main/java/net/fabricmc/fabric/api/resource/SimpleResourceReloadListener.java]: I'll fix them before merging

**Comment by modmuss50** [comment]
Looks good, I like the idea of making the `SimpleResourceReloadListener`, as the standard api could confuse people (including me),

Name wise I cannot really think of anything better. The other changes (packet renames mainly) look to be the same as what I did.

**Comment by asiekierka** [comment]
https://github.com/FabricMC/fabric/commit/377a882f62af04c0838532daf1cbaf5ac83d08e9

**Pull request closed without merging**